### PR TITLE
Remove NSFV Action and NSFV component, use Physics

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #25642

### CODE REVIEW

**Comment by lindsayad** [comment]
Are the TH area users ready for this?

**Comment by GiudGiud** [comment]
@licharlot @tanoret

**Comment by GiudGiud** [comment]
Note that the old syntax remains completely functional. It's just the backend that changes (towards something that uses mostly the same code, just organized differently).
You do get a deprecation message.

**Comment by GiudGiud** [comment]
Public app failure is fake

**Comment by lindsayad** [comment]
> Note that the old syntax remains completely functional. It's just the backend that changes (towards something that uses mostly the same code, just organized differently).
> You do get a deprecation message.

Oh nice 👍

File: modules/navier_stokes/test/tests/finite_volume/ins/boussinesq/tests [hunk 1]
```diff
@@ -46,11 +46,13 @@
   [rho-error]
     type = RunException
     input = boussinesq-action.i
-    cli_args = "Modules/NavierStokesFV/density='x'"
+    cli_args = "Modules/NavierStokesFV/density='rho_error'"
     expect_err = 'rho: The density in the boussinesq term is not constant!'
     requirement = 'The system should throw an error if the density is not a constant functor in case of Boussinesq treatment.'
     valgrind = 'none'
     issues = '#19742'
+    # New physics cannot support this because we want properties forwarded from fluid properties to work
```
Comment (joshuahansel) [inline modules/navier_stokes/test/tests/finite_volume/ins/boussinesq/tests:54]: I don't quite understand. So the Physics version is not going to take density as a functor?

**Review by joshuahansel — COMMENTED** [review]
I want to discuss

> Had to regold because using a single physics for both components and the two components were using different advection methods. This is not one of the block restricted parameters.
The other solution would have been to use two physics. This would necessitate different variable names at the moment. There were some work on merging physics together
It is possible, it's also quite a bit of code to achieve something that so far is not really needed

In particular the part about needing different variables if needing more than one physics. I just want to understand any limitations before tighter integration between components and physics. But the change to this test is probably fine.

**Comment by GiudGiud** [comment]
Unless we start changing the block restrictions of created objects, I don't see how we could not use different variable names. 

The physics create block restricted variables and UOs

File: modules/navier_stokes/test/tests/finite_volume/ins/boussinesq/tests (hunk 1 above)
Comment (joshuahansel) [reply modules/navier_stokes/test/tests/finite_volume/ins/boussinesq/tests:54]: So how does Physics affect this? And does this test make sense to exist? Do we just need to improve `isConstant` somehow?

Comment (GiudGiud) [reply modules/navier_stokes/test/tests/finite_volume/ins/boussinesq/tests:54]: >  Do we just need to improve isConstant somehow?

I dont see an easy way. Save for maybe evaluating it all over the mesh to check.

> So how does Physics affect this?

I dont want to limit incompressible flow to not using the FP properties and the GFFP. I ll give it another try to see if I can re-enable this.

**Comment by joshuahansel** [comment]
Ok, I'm good with these changes, so just checking for any dissent by @licharlot or @tanoret before approval.

**Comment by GiudGiud** [comment]
let's go?

**Comment by licharlot** [comment]
Looks good to me. Thanks for adding the transition tutorial.

File: modules/navier_stokes/src/actions/NSFVActionDeprecation.C [hunk 2]
```diff
@@ -25,7 +25,9 @@ NSFVActionDeprecation::NSFVActionDeprecation(const InputParameters & parameters)
 {
   mooseDeprecated(
       "The NSFVAction, used through the [Modules/NavierStokesFV] syntax is deprecated. "
-      "Please convert your input file to using the [Physics/NavierStokes/WCNSFV...] syntax.");
+      "Please convert your input file to using the [Physics/NavierStokes/WCNSFV...] syntax.\n"
```
Comment (joshuahansel) [inline modules/navier_stokes/src/actions/NSFVActionDeprecation.C:28]:
```suggestion
      "Please modify your input file to use the [Physics/NavierStokes/*] syntax.\n"
```

Comment (joshuahansel) [reply modules/navier_stokes/src/actions/NSFVActionDeprecation.C:28]: I see much or all of the syntax doesn't start with "WCNSFV"

File: modules/navier_stokes/doc/content/syntax/Modules/NavierStokesFV/index.md [hunk 3]
```diff
@@ -149,6 +152,50 @@ energy source function can also be supplied to the incorporated
...
 
 
+## How to transition to the Physics syntax id=transition
+
+The `/Modules/NavierStokes` action syntax and (`Physics/NavierStokes/Flow/..`, `Physics/NavierStokes/FluidHeatTransfer/..`, `Physics/NavierStokes/ScalarTransport/..`, `Physics/NavierStokes/Turbulence/..`) syntax
+create the exact same objects in the background. We currently do not expect any difference in results,
+notably because `/Modules/NavierStokes` has been changed to create the relevant `Physics` under the hood!
+
+To transition, you will have to split the `/Modules/NavierStokes` parameters into four groups below.
+Your simulation may only feature only one of these groups, in which case you will only need to create a single `Physics`.
```
Comment (joshuahansel) [inline modules/navier_stokes/doc/content/syntax/Modules/NavierStokesFV/index.md:162]:
```suggestion
Your simulation may only feature only one of these groups, in which case you will only need to create a single `Physics`:
```

File: modules/navier_stokes/doc/content/syntax/Modules/NavierStokesFV/index.md [hunk 4]
```diff
@@ -149,6 +152,50 @@ energy source function can also be supplied to the incorporated
...
+the scalar conservation equations parameters into this `[Physics/NavierStokes/ScalarTransport/<name>]` syntax:
+
+!listing finite_volume/ins/channel-flow/2d-scalar-transport-physics.i start=Physics/NavierStokes end=[AuxVariables]
+
+and finally the turbulence parameters into this `[Physics/NavierStokes/Turbulence/<name>]` syntax:
+
+!listing finite_volume/ins/channel-flow/2d-mixing-length-physics.i start=Physics/NavierStokes end=[Executioner]
+
+!alert note
+All `NavierStokes` `Physics` and be nested under the same `[NavierStokes]` sub-block. The separation in the examples into different `Physics` blocks is only for the purpose of simplifying the examples.
```
Comment (joshuahansel) [inline modules/navier_stokes/doc/content/syntax/Modules/NavierStokesFV/index.md:187]:
```suggestion
All `NavierStokes` `Physics` may be nested under the same `[NavierStokes]` sub-block. The separation in the examples into different `Physics` blocks is only for the purpose of simplifying the examples.
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Minor changes. For the transition doc, I think it would be useful to have 2 columns of code listing for the example - the before and after conversion for your example. And you could keep all of the physics in the same block and just discuss all afterward. But this is an optional change.

**Comment by GiudGiud** [comment]
Can we merge this? 
If this triggers downstream VTB failure I'd like to be able to address them over the 4th

**Review by joshuahansel — APPROVED** [review]
Looks good. Sorry for the wait; I didn't notice you were ready for review again.

**Comment by GiudGiud** [comment]
Thanks for the review! One last PR before being done with what was reported on the M2

**Pull request merged**